### ISSUE TITLE
[DbDataStore] repair is very slow on big tables

### ISSUE THREAD

**Issue by cdealti** [eclipse/kura#752]
The ``repair`` operation contains a DB query to detect duplicates of the primary key (which should not occur).
The operation was added due to this [bug](https://sourceforge.net/p/hsqldb/discussion/73674/thread/a08046eb/).

The query has shown to take a very long time (in the order of minutes) on relatively large **file** databases (for example enabling the "almost persistent" configuration in kura.properties).

The bug is only known to happen on certain gateways using a raw flash storage configuration.

Since these gateways are no longer supported, the ``repair`` could be executed only once on startup maybe subjected to a configuration parameter.

**Comment by cdealti** [comment]
This will be addressed as part of the switch to the H2 Database.

**Comment by nicolatimeus** [comment]
The db repair addresses an HSQL-specific issue so hopefully it is no longer needed after the switch to H2.
The code that performs the repair is still present in the H2 implementation but is disabled by default. It can be re-enabled setting `db.store.repair.enabled=true` in kura.properties.
